src/crimson/net/ProtocolV1.cc

```diff
@@ -25,6 +25,26 @@
 WRITE_RAW_ENCODER(ceph_msg_connect);
 WRITE_RAW_ENCODER(ceph_msg_connect_reply);
 
+namespace {
+template<typename T, typename=void>
+struct unaligned_t;
+
+template<typename T>
+struct unaligned_t<T, enable_if_t<std::is_pod_v<T>>>
+{
+  T value;
+} __attribute__((packed));
+
+template<typename T>
+inline T unaligned_load(const char* p)
```

Comment:
Does it also applies to `ProtocolV2.cc`? such as:
* https://github.com/ceph/ceph/blob/master/src/crimson/net/ProtocolV2.cc#L207
* https://github.com/ceph/ceph/blob/master/src/crimson/net/ProtocolV2.cc#L289

Reply:
sure. i was testing with v1 protocol. will fix v2 as well in the next revision.

Reply:
@cyx1231st i withdraw my words. the accessed structs in v2 are packed. so they are immune to the unaligned access issue.